crates/aptos/src/account/key_rotation.rs

```diff
@@ -0,0 +1,259 @@
+            ..self.txn_options.profile_options.profile()?
+        };
+
+        let mut config = CliConfig::load(ConfigSearchMode::CurrentDirAndParents)?;
+
+        if config.profiles.is_none() {
+            // This should not happen. The command requires a profile exist to rotate key
+            config.profiles = Some(BTreeMap::new());
+        }
+        config
+            .profiles
+            .as_mut()
+            .unwrap()
+            .insert(profile_name.clone(), profile_config);
+        config.save()?;
+
+        eprintln!("Profile {} is saved.", profile_name);
+
+        Ok(())
+    }
+}
+
+/// Command to lookup the account adress through on-chain lookup table
+///
+#[derive(Debug, Parser)]
+pub struct LookupAddress {
+    #[clap(flatten)]
+    pub(crate) encoding_options: EncodingOptions,
+
+    #[clap(flatten)]
+    pub(crate) public_key_options: PublicKeyInputOptions,
+
+    #[clap(flatten)]
+    pub(crate) profile_options: ProfileOptions,
+
+    #[clap(flatten)]
+    pub(crate) rest_options: RestOptions,
+}
+
+impl LookupAddress {
+    pub(crate) fn public_key(&self) -> CliTypedResult<Ed25519PublicKey> {
+        self.public_key_options.extract_public_key(
+            self.encoding_options.encoding,
+            &self.profile_options.profile,
+        )
+    }
+
+    /// Builds a rest client
+    fn rest_client(&self) -> CliTypedResult<Client> {
+        self.rest_options.client(&self.profile_options.profile)
+    }
+}
+
+#[async_trait]
+impl CliCommand<String> for LookupAddress {
+    fn command_name(&self) -> &'static str {
+        "LookupAddress"
+    }
+
+    async fn execute(self) -> CliTypedResult<String> {
+        let originating_resource = self
+            .rest_client()?
+            .get_account_resource(CORE_CODE_ADDRESS, "0x1::account::OriginatingAddress")
+            .await
+            .map_err(|err| CliError::ApiError(err.to_string()))?
+            .into_inner()
+            .ok_or_else(|| CliError::UnexpectedError("Unable to parse API response.".to_string()))?
+            .data;
+
+        let table_handle = originating_resource["address_map"]["handle"]
+            .as_str()
+            .ok_or_else(|| {
+                CliError::UnexpectedError("Unable to parse table handle.".to_string())
+            })?;
+
+        // The derived address that can be used to look up the original address
+        let address_key = AuthenticationKey::ed25519(&self.public_key()?).derived_address();
+
+        Ok(self
+            .rest_client()?
+            .get_table_item(
+                table_handle,
+                "address",
+                "address",
+                address_key.to_hex_literal(),
+            )
+            .await
+            .map_err(|err| CliError::ApiError(err.to_string()))?
+            .into_inner()
+            .as_str()
+            .ok_or_else(|| CliError::UnexpectedError("Unable to parse API response.".to_string()))?
+            .to_string())
```

Comment:
You shouldn't have to stringify this, just leave it as the raw JSON object.  What does it look like before .to_string().

I see, this return type should be `AccountAddress` right?